ipn/ipnlocal/local.go

```diff
@@ -5041,6 +5057,15 @@ func (b *LocalBackend) HandleQuad100Port80Conn(c net.Conn) error {
 	return s.Serve(netutil.NewOneConnListener(c, nil))
 }
 
+// handleWebClientConn serves web client requests.
+func (b *LocalBackend) handleWebClientConn(c net.Conn) error {
+	if err := b.WebInit(); err != nil {
+		return err
+	}
+	s := http.Server{Handler: b.web.ws}
+	return s.Serve(netutil.NewOneConnListener(c, nil))
```

Comment:
If we initialize a new web server on each connection, we'll loose our session state each time, right? I wonder if instead we should just initialize it when the pref gets updated.

Reply:
we're not creating a new `client/web.Server` on each request, just a new `http.Server`.  The session state is held in `b.web.ws` which is reused between requests.

But even recreating the `http.Server` on every request seems a little wasteful, and could probably be optimized.  That said, it's what we're also doing for tailscale serve, peerapi, etc, so it must not be too bad.

Reply:
Oh I see! Because `WebInit` doesn't re-initialize if already initialized.